python/ray/data/tests/test_file_based_datasource.py

```diff
@@ -129,6 +132,47 @@ def test_valid_shuffle_arg_does_not_raise_error(ray_start_regular_shared, shuffl
     FileBasedDatasource("example://iris.csv", shuffle=shuffle)
 
 
+def test_read_parquet_deterministic_shuffling():
+    def write_parquet_file(path, file_index):
+        """Write a dummy Parquet file with test data."""
+        # Create a dummy dataset with unique data for each file
+        data = {
+            "col1": range(10 * file_index, 10 * (file_index + 1)),
+            "col2": ["foo", "bar"] * 5,
+        }
+        table = pyarrow.Table.from_pydict(data)
+        pq.write_table(table, path)
+
+    ctx = ray.data.DataContext.get_current()
+    ctx.execution_options.preserve_order = True
+    # Use the current working directory
+    current_dir = Path(os.getcwd())
```

Comment:
Instead of using the current directory and a try-finally for cleanup, use the tmp_path pytest fixture. https://docs.pytest.org/en/stable/how-to/tmp_path.html